helm_image_mapping.py

```
@@ -0,0 +1,91 @@
+) -> dict:
+    for resource in component.resources:
+        if resource_name != resource.name:
+            continue
+        if resource_version and resource_version != resource.version:
+            continue
+        if mapping_type != resource.type:
+            continue
+        if resource_extra_id and resource.extraIdentity != resource_extra_id:
+            continue
+        break
+    else:
+        raise ValueError(f'Error: did not find {mapping_type=} for {resource=}')
+
+    if not resource.access.type is ocm.AccessType.LOCAL_BLOB:
+        raise ValueError(f'{component.name}/{resource.name} has unexpected {resource.access.type=}')
+
+    oci_ref = component.current_ocm_repo.component_version_oci_ref(component)
+
+    image_mappings = oci_client.blob(
+        image_reference=oci_ref,
+        digest=resource.access.localReference,
+        stream=False, # imagemaps are typically small, so it should be okay to read into memory
+    ).json()['imageMapping']
+
+    if not isinstance(image_mappings, list):
+        cv = f'{component_name}:{component_version}'
+        raise ValueError(f'imagemapping of {cv} does not match expected format')
+
+    values = {}
+    for image_mapping in image_mappings:
+        # image-mapping is expected to contain the following attributes:
+        #
+        # resource:
+        #   name: oci-image-resource-name (for looking up image-resource)
+        # repository: <attribute-name to set resource's image-repository to>
+        # tag: <attribute-name to set resource's image-tag to
+
+        resource_name = image_mapping['resource']['name']
+        for resource in component.resources:
+            if resource.name != resource_name:
+                continue
+            if not resource.type is ocm.ArtefactType.OCI_IMAGE:
+                continue
+            break # found it
+        else:
+            print(f'Error: did not find oci-image w/ {resource_name=} in {component=}')
+            exit(1)
```

ditto: Raise instead of print + exit.